### PR TITLE
spec and implement variable saltpack header branding

### PR DESCRIPTION
**Pull request by oconnor663**
@maxtaco @patrickxb

### CODE REVIEW

File: go/libkb/saltpack.go [hunk 1]
```diff
@@ -138,8 +138,6 @@ func BoxPublicKeyToKeybaseKID(k saltpack.BoxPublicKey) (ret keybase1.KID) {
 }
 
 func checkSaltpackBrand(b string) error {
-	if b != KeybaseSaltpackBrand {
-		return KeybaseSaltpackError{}
-	}
```
Comment (oconnor663) [inline go/libkb/saltpack.go]: I remember that Patrick and I wanted to delete this, but Max wanted to keep it. If we keep this check, then the example blob in the spec below will need to include the KEYBASE brand, or else we won't be able to verify it. I don't feel super strongly either way.

Comment (patrickxb) [inline go/libkb/saltpack.go]: I feel like as this is the first saltpack implementation released into the wild, that it should accept any brand.

If I write a version in haskell and want to test that it works, I would use the keybase CLI to check the messages I generated.  And with this in place, I would have to use KEYBASE brand...

Perhaps decrypt, verify could display the brand?

Comment (maxtaco) [inline go/libkb/saltpack.go]: if it's 2 against 1, i relent...

**Comment by oconnor663** [comment]
5299985301b8a194c090dada53ea12b9e2f49e7a

**Pull request closed without merging**